File: Katiba55.API/Entities/Officer.cs
namespace Katiba55.API.Entities
{
    public class Officer: BaseEntity
    {
        public string Name { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public OfficerRanks Rank { get; set; }
        public OfficerStatus Status { get; set; }
        public DateTime? JoinDate { get; set; }
        public DateTime? LeaveDate { get; set; }
        public string? Notes { get; set; }

        public ICollection<Project> Projects { get; set; }
    }
}

File: Katiba55.Client/src/app/models/officers/officer.ts
import { OfficerRank } from "../../enums/officer-rank.enum"
import { OfficerStatus } from "../../enums/officer-status.enum"

export interface Officer {
    id: number
    name: string
    email?: string
    phone?: string
    rank: OfficerRank
    status: OfficerStatus
    joinDate?: Date
    leaveDate?: Date
    notes?: string
}
